File: Swift/프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.swift
func solution(_ n:Int64) -> Int64 {
    var N = n
    var intArr: [Int64] = []
    while N > 0 {
        intArr.append(N%10)
        N /= 10
    }
    
    intArr.sort(by: { $0 < $1 })
    var result: Int64 = 0
    var mul: Int64 = 1
    for element in intArr {
        result += element * mul
        mul *= 10
    }
    return result
}

File: Java/프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
class Solution {
    public long solution(long n) {
        int[] arr = new int[10];
        while (n > 0) {
            arr[(int)(n%10)]++;
            n /= 10;
        }
        
        long answer = 0;
        for (int i = 9; i >= 0; i--) {
            for (int j = 0; j<arr[i]; j++) {
                answer += i;
                answer *= 10;
            }
        }
        return answer / 10;
    }
}